test/util/util.go

```diff
@@ -140,25 +140,24 @@ func DeleteAllPodsInNamespace(ctx context.Context, c client.Client, ns *corev1.N
 		return fmt.Errorf("deleting all Pods in namespace %q: %w", ns.Name, err)
 	}
 
-	lst := corev1.PodList{}
-	err = c.List(ctx, &lst, client.InNamespace(ns.Name))
-	if err != nil && !apierrors.IsNotFound(err) {
-		return fmt.Errorf("listing Pods with a finalizer in namespace %q: %w", ns.Name, err)
-	}
-
-	if len(lst.Items) == 0 {
-		return nil
-	}
+	gomega.Eventually(func() error {
+		lst := corev1.PodList{}
+		err := c.List(ctx, &lst, client.InNamespace(ns.Name))
+		if err != nil && !errors.IsNotFound(err) {
+			return fmt.Errorf("listing Pods with a finalizer in namespace %q: %w", ns.Name, err)
```

Comment:
Although I believe that we can improve error logging by using `Any()` and `func(g gomega.Gomega)`, we can work on such a cleanup at another PR.

Reply:
I thought about it, but it we still need to check `IsNotFound` in the loop below, so I decided against it.